sdlib/d/gc/tcache.d

```diff
@@ -413,11 +413,25 @@ private:
 
 		if (allocated >= nextAllocationEvent) {
 			recycleNextBin();
+			// check for triggering an automatic collection
+			checkAutomaticCollection();
```

Comment:
It does not seems reasonable to check for this every 64k when a block is 2MB and collection triggers at the block level.